File: api/airport_manager/src/main/java/com/awbd/airport_manager/service/api/GateService.java
package com.awbd.airport_manager.service.api;

import com.awbd.airport_manager.dto.GateDto;
import com.awbd.airport_manager.util.pagination.PagedResponse;
import com.awbd.airport_manager.util.search.dto.SearchDTO;

import java.util.UUID;

public interface GateService {

    PagedResponse<GateDto> search(SearchDTO searchDTO);

    GateDto getById(UUID id);

    GateDto create(GateDto dto);

    GateDto update(UUID id, GateDto dto);

    void delete(UUID id);
}

File: ui/src/app/core/services/gate.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Gate, GateForm } from '../../shared/models/gate.model';
import { PagedResponse } from '../../shared/models/paged-response.model';
import { SearchDTO } from '../../shared/models/search.model';

@Injectable({ providedIn: 'root' })
export class GateService {
  private http = inject(HttpClient);
  private base = `${environment.apiUrl}/api/gates`;

  search(query: SearchDTO): Observable<PagedResponse<Gate>> {
    return this.http.post<PagedResponse<Gate>>(`${this.base}/search`, query);
  }

  getById(id: string): Observable<Gate> {
    return this.http.get<Gate>(`${this.base}/${id}`);
  }

  create(payload: GateForm): Observable<Gate> {
    return this.http.post<Gate>(this.base, payload);
  }

  update(id: string, payload: GateForm): Observable<Gate> {
    return this.http.put<Gate>(`${this.base}/${id}`, payload);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
